### PR TITLE
fix: `Selections` arrow overlap borders in bordered tables

### LINKED ISSUE #29083: `Selections` arrow overlap borders in bordered tables
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://hmzhe.csb.app/](https://codesandbox.io/s/ghfl4?file=/index.js)

### Steps to reproduce
1. enable selection actions in table
2. add border to table

### What is expected?
no overlap, wider paddings

### What is actually happening?
arrow overlaps border

| Environment | Info |
|---|---|
| antd | 4.11.2 |
| React | 17 |
| System | mac os |
| Browser | chrome 88 |

image

### PR DESCRIPTION
**Pull request by duuliy**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #29083

### 💡 Background and solution

| before   | after |
| ---------- | --------- |
| WX20210130-151254@2x |     WX20210130-023247@2x  |

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     When the table has selections, the width of the selected column is restored to the previous one, otherwise it is 32px      |
| 🇨🇳 Chinese |    当 Table 有selections时选中列宽度恢复到之前宽度，否则为 32px。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
Changelog 完善下

**Comment by zombieJ** [comment]
Use flex layout instead of hard code style.

File: components/table/hooks/useSelection.tsx [hunk 1]
```diff
@@ -589,7 +589,7 @@ export default function useSelection<RecordType>(
         title: rowSelection.columnTitle || title,
         render: renderSelectionCell,
         [INTERNAL_COL_DEFINE]: {
-          className: `${prefixCls}-selection-col`,
+          className: `${prefixCls}-${selections ? 'selections-col' : 'selection-col'}`,
```
Comment (zombieJ) [inline components/table/hooks/useSelection.tsx:592]: add new class name like `xxx-with-selections`?

**Comment by duuliy** [comment]
Why does Argos fail

File: components/table/hooks/useSelection.tsx (hunk 1 above)
Comment (duuliy) [reply components/table/hooks/useSelection.tsx]: well. Why did Argos fail

File: components/table/hooks/useSelection.tsx [hunk 2]
```diff
@@ -589,7 +589,7 @@ export default function useSelection<RecordType>(
         title: rowSelection.columnTitle || title,
         render: renderSelectionCell,
         [INTERNAL_COL_DEFINE]: {
-          className: `${prefixCls}-selection-col`,
+          className: `${prefixCls}-${selections ? 'with-selections-col' : 'selection-col'}`,
```
Comment (afc163) [inline components/table/hooks/useSelection.tsx:592]: 有 border 的时候才恢复吧。

Comment (afc163) [reply components/table/hooks/useSelection.tsx:592]: ping @duuliy

**Pull request closed without merging**